posthog/hogql/visitor.py

```diff
@@ -247,6 +247,9 @@ def visit_window_frame_expr(self, node: ast.WindowFrameExpr):
     def visit_join_constraint(self, node: ast.JoinConstraint):
         self.visit(node.expr)
 
+    def visit_expression_field_type(self, node: ast.ExpressionFieldType):
+        self.visit(node.expr)
+
```

Comment:
I wonder if this is correct. In the resolver, [when we resolve expression fields](https://github.com/PostHog/posthog/blob/c6bb217c80532ab599983124ef930744a2dd999e/posthog/hogql/resolver.py#L501-L507), we clone this `node.expr` and swap out the node with ExpressionFieldType with the cloned expr.

The change here probably has some side effects, and/or might mutate the "reference expression" instead of cloning and applying. Not sure, but it might be worth double checking.

Reply:
Interesting, yeah, some tests were failing due to `TraversingVisitor` not having a `visit_expression_field_type` func - we do `pass` on some, which may be better here than visiting the expression